# Q-8.
The anxiety levels of 10 participants were measured before and after a new therapy. The scores are not normally distributed. Use the Wilcoxon signed-rank test to test whether the therapy had a significant effect on anxiety levels. 

The data is given below: Participant Before therapy After therapy Difference:

![data](data_Q8.png)

In [5]:
import pandas as pd
import scipy.stats as stats

In [4]:
df = pd.DataFrame({'Participant':[1,2,3,4,5,6,7,8,9,10],
                  'Before_therapy':[10,8,12,15,6,9,11,7,14,10],
                  'After_therapy':[7,6,10,12,5,8,9,6,12,8]})
df['Difference'] = df.Before_therapy - df.After_therapy

df

Participant  Before_therapy  After_therapy  Difference
0            1              10              7           3
1            2               8              6           2
2            3              12             10           2
3            4              15             12           3
4            5               6              5           1
5            6               9              8           1
6            7              11              9           2
7            8               7              6           1
8            9              14             12           2
9           10              10              8           2

In [7]:
statistic, p_value = stats.wilcoxon(df.Difference)

print("Wilcoxon signed-rank test:")
print("Statistic:", statistic)
print("p-value:", p_value)

Wilcoxon signed-rank test:
Statistic: 0.0
p-value: 0.001953125


Assuming significance level of 0.05, since the p-value (0.001953125) is less than 0.05, we can conclude that there is strong evidence to *reject the null hypothesis.* This means that the therapy had a significant effect on anxiety levels.

In other words, the results suggest that there is a **significant difference between the anxiety levels before and after the therapy.**